spec/spec_helper.rb

```diff
@@ -114,11 +115,23 @@ class Pet < ActiveRecord::Base
   col :home_address, :type => :text
   if ENV['DB'] == 'postgresql'
     col :tsntz, :type => 'timestamp without time zone'
+  else
+    add_index :name, :unique => true
+  end
+end
+if ENV['DB'] == 'postgresql'
+  begin
+    Pet.connection.execute("ALTER TABLE pets DROP CONSTRAINT IF EXISTS unique_name")
+  rescue => e
+    puts e.inspect
   end
-  add_index :name, :unique => true
 end
+
 Pet.auto_upgrade!
 
+if ENV['DB'] == 'postgresql'
```

Comment:
similar except that it means people adding tests in the future always have to write 2 versions

ideally, most generic and postgres-specific tests (except the ones that explicitly check native or non-native things) run on all possible combinations

pseudocode:

```
if ENV['DB'] == 'postgres' and ENV['UNIQ_INDEX'] == 'true'
  # add uniq constraint
end

test "foobar" do
  upsert = Upsert.new [...] disable_native: (ENV['DISABLE_NATIVE'] == 'true')
  [...]
end
```